src/middleware/requireAuthentication.ts

```diff
@@ -0,0 +1,17 @@
+import { Response, NextFunction } from 'express';
+import { UnauthorizedError } from '../errors';
+import type { Request as JWTRequest } from 'express-jwt';
+
+const requireAuthentication = (
+	req: JWTRequest,
+	res: Response,
+	next: NextFunction,
+) => {
+	if (req.auth === undefined) {
+		next(new UnauthorizedError('No authorization token was found'));
```

Comment:
What is the difference between calling `next` with an `Error` versus throwing an `Error`? What is the advantage of the former?

Reply:
Throwing an error crashes the process / crashes the server.

`next` passes the error to the error handling middleware (which then translates it into an HTTP status code, in this case a 401)